gemd/entity/object/has_template_check_generator.py

```diff
@@ -0,0 +1,71 @@
+"""For entities that have specs."""
+from gemd.entity.template.base_template import BaseTemplate
+from gemd.entity.link_by_uid import LinkByUID
+from gemd.entity.bounds_validation import get_validation_level, WarningLevel
+from gemd.entity.dict_serializable import logger
+
+from abc import ABC, abstractmethod
+from inspect import getmodule, getmembers, isclass, signature
+from typing import Union, Callable, TypeVar
+
+T = TypeVar('T')
+
+
+class HasTemplateCheckGenerator(ABC):
+    """Mix-in trait for objects that can generate a Check Template closure."""
+
+    @property
+    @abstractmethod
+    def template(self) -> Union[BaseTemplate, LinkByUID]:
+        """Get the object template associated with this object."""
+
+    def _generate_template_check(self,
```

Comment:
Can you add in this docstring a comment describing the signature of the returned closure function please? The comment `Generate a closure for the object and the validation routine.` is not as clear as one would hope.

E.g. "The generated closure references the class type that calls this method, and takes as argument a value to validate. The value is checked against blah blah..."